### ISSUE TITLE
Feature Request: Option to use HTTP Proxies

### ISSUE THREAD

**Issue by ishaanpathak** [Beastwick18/nyaa#11]
So, for the issue of not being able to open nyaa.si, there is one more way of accessing it, which is to use a tool that Evades DPI (Deep Packet Inspection). But those tools usually work like http proxies on the system so maybe have an option to be able to use custom http proxies? Could be a niche use case though. Once such tool that I personally use is this: https://github.com/xvzc/SpoofDPI

It pretty much unblocks all of the blocked websites for me. As far as I know, my ISP just uses DPI to stop the websites. But I don't think many people would be using this tool so if you do consider this feature, just keep it on low priority because the project is so far great!

**Comment by Beastwick18** [comment]
Commit 6077afd *should* add support for this. Would it be possible for you to build from source and let me know if proxies are working? I've tested it on Privoxy, which is what I use, and it seems to be working.

To get it working, add the line:
```toml
request_proxy = "127.0.0.1:1234"
```
and fill in your own proxy url and port.

**Comment by ishaanpathak** [comment]
Just tried it out, and it works! Thanks a lot!

**Comment by Beastwick18** [comment]
Awesome, this will be in the next release then :)
